File: backend_v2/app/core/providers/provider_factory.py
import os

# from app.core.providers.openai_provider import OpenAIProvider
from app.core.providers.google_provider import GoogleProvider
from app.core.providers.local_embedding_provider import LocalEmbeddingProvider


def get_ai_provider():
    provider_name = os.getenv("AI_PROVIDER", "google").lower()

    if provider_name == "google":
        return GoogleProvider()
    # elif provider_name == "openai":
    #     return OpenAIProvider()
    else:
        raise ValueError(f"Provider de IA desconhecido: {provider_name}")


def get_embedding_provider():
    provider_name = os.getenv("EMBEDDING_PROVIDER", "google").lower()

    return LocalEmbeddingProvider()
    if provider_name == "google":
        return GoogleProvider()
    # elif provider_name == "openai":
    #     return OpenAIProvider()
    else:
        raise ValueError(f"Provider de Embeddings desconhecido: {provider_name}")

File: backend/chat/src/core/providers/providerFactory.ts
import GoogleProvider from './GoogleProvider';

function getAIProvider() {
  const providerName = (process.env.AI_PROVIDER || 'google').toLowerCase();

  if (providerName === 'google') {
    return new GoogleProvider();
  }
  throw new Error(`Provider de IA desconhecido: ${providerName}`);
}

function getEmbeddingProvider() {
  const providerName = (process.env.EMBEDDING_PROVIDER || 'google').toLowerCase();

  if (providerName === 'google') {
    return new GoogleProvider();
  }
  throw new Error(`Provider de Embeddings desconhecido: ${providerName}`);
}

export { getAIProvider, getEmbeddingProvider };

